pkg/controller/amazoncloudintegration/amazoncloudintegration_controller.go

```diff
@@ -187,7 +187,7 @@ func (r *ReconcileAmazonCloudIntegration) Reconcile(request reconcile.Request) (
 
 	// Render the desired objects from the CRD and create or update them.
 	reqLogger.V(3).Info("rendering components")
-	component, err := render.AmazonCloudIntegration(instance, network, awsCredential, pullSecrets, r.provider == operatorv1.ProviderOpenShift)
+	component, err := render.AmazonCloudIntegration(instance, network.Spec, awsCredential, pullSecrets, r.provider == operatorv1.ProviderOpenShift)
```

Comment:
Seems like you've created a new type `InstallationInternal` which allows you to pass both the spec and the additional annotation options via a single struct. I like this idea, but I think it belongs in the render package as a `RenderAmazonCloudOptions` struct. And we could even move all these other arguments into it too, over time. I've always thought this could work well for us since if you look in our tests, we usually pass the default empty values for most of these anyways (false, nil, "", etc.).

Alternatively, we could follow the existing pattern by just passing those additional options (tigeracustom / nodeapparmor) as additional args to the render functions?

Reply:
We chatted about this in slack and I think we're good with the InstallationInternal for now.